catkin_ws/src/controls/src/servers/state_quaternion_pid_server.py

```diff
@@ -50,9 +52,14 @@ def callback(self, goal):
         my_goal = self.goal_id
         self.cancelled = False
         self.goal = goal
+        
         if self.pose is not None:
             goal_position = [self.goal.pose.position.x, self.goal.pose.position.y, self.goal.pose.position.z]
             goal_quat = np.quaternion(self.goal.pose.orientation.w, self.goal.pose.orientation.x, self.goal.pose.orientation.y, self.goal.pose.orientation.z)
+
+            self.buoyant_force_pub.publish(Bool(True))
```

Comment:
code looks good, I'd just reverse the logic for safety (buoyant force is off when a pose is received, and only turned on if there is a Z position goal which is below the z threshold). I think the way it is written now, even if the goal does not contain a Z position target (for example it just sets the Y position or the quaternion), the buoyancy force will activate, even though the AUV's Z position might not be below the threshold.